### PR TITLE
Release automation addedd for flyte definition 

### PR DESCRIPTION
**Pull request by evalsocket**
Added on-demand Github workflow that runs the shell script, that updates all flyte component then runs make kustomize then opens a PR with those changes.

### CODE REVIEW

**Comment by katrogan** [comment]
@evalsocket this is so awesome, thank you!!!

File: .github/workflows/release-kustomize.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        run: |
+          make release_automation
+          make kustomize
+      - name: Create Pull Request
+        id: cpr
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.FLYTE_BOT_PAT }}
+          commit-message: Updated flyte defination
+          committer: GitHub <flytebot@flyte.org>
```
Comment (EngHabu) [inline .github/workflows/release-kustomize.yml:24]:
```suggestion
          committer: Flyte-Bot <admin@flyte.org>
```

File: .github/workflows/release-kustomize.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          signoff: true
+          branch: bot-flyte-defination
```
Comment (EngHabu) [inline .github/workflows/release-kustomize.yml:27]:
```suggestion
          branch: flyte-bot-update-releases
```

File: .github/workflows/release-kustomize.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+          delete-branch: true
+          title: 'Updated flyte defination'
```
Comment (EngHabu) [inline .github/workflows/release-kustomize.yml:29]:
```suggestion
          title: 'Update Flyte components' releases'
```

File: .github/workflows/release-kustomize.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          body: |
+            Updated flyte defination
+            - Updated GCP Flyte defination
+            - Updated EKS Flyte defination
+            - Updated SANDBOX Flyte defination
+            - Updated TEST Flyte defination
+            - Auto-generated by [flytebot]
```
Comment (EngHabu) [inline .github/workflows/release-kustomize.yml:36]:
```suggestion
            Updated flyte deployment
            - Updated GCP Flyte deployment
            - Updated EKS Flyte deployment
            - Updated Sandbox Flyte deployment
            - Updated TEST Flyte deployment

            - Auto-generated by [flyte-bot]
```

File: Makefile [hunk 5]
```diff
@@ -3,8 +3,13 @@ pip-compile $(1) --upgrade --verbose
 endef
 
 .PHONY: kustomize
-kustomize:
+kustomize: 
 	KUSTOMIZE_VERSION=3.9.2 bash script/generate_kustomize.sh
+	git stash push -- kustomize/
```
Comment (EngHabu) [inline Makefile:8]: What does this do?

File: kustomize/overlays/test/kustomization.yaml [hunk 6]
```diff
@@ -19,23 +19,19 @@ bases:
...
+    newTag: v0.3.1 # DATACATALOG_TAG override the tag
     newName: ghcr.io/flyteorg/datacatalog # override the name
   # FlytePropeller
   - name: flytepropeller # match images with this name
-    newTag: v0.7.8 # override the tag
-    newName: ghcr.io/flyteorg/flytepropeller # override the name
-  # Webhook
-  - name: webhook # match images with this name
-    newTag: v0.7.8 # override the tag
+    newTag: v0.7.8 # FLYTEPROPELLER_TAG override the tag
```
Comment (EngHabu) [inline kustomize/overlays/test/kustomization.yaml:34]: See on the left? a new image is added (webhook) it uses the same flytepropeller image... looks like the master merge here didn't happen properly

**Review by EngHabu — APPROVED** [review]

**Pull request merged**